File: code/test_suites/cpp/FunctionBuildTest/tests/test_catboost_predict.cpp
#include <gtest/gtest.h>
#include "../src/function_catboost_predict.cpp"

class TestBoostPredictor : public ::testing::Test {
protected:
    void SetUp() override {
        predictor = new BoostPredictor();
    }

    void TearDown() override {
        delete predictor;
    }

    BoostPredictor* predictor;
};

TEST_F(TestBoostPredictor, TestBasicClassification) {
    int result = predictor->predict_class(1, 2, 3, 4, 5, 0, 0, 1, 1, 1, 3);
    ASSERT_EQ(result, 1);
}

TEST_F(TestBoostPredictor, TestCustomThreshold) {
    int result = predictor->predict_class(10, 15, 20, 25, 30, 0, 0, 1, 1, 1, 18);
    ASSERT_EQ(result, 1);
}

TEST_F(TestBoostPredictor, TestEdgeCaseLowValue) {
    int result = predictor->predict_class(5, 6, 7, 8, 9, 0, 0, 1, 1, 1, 4);
    ASSERT_EQ(result, 0);
}

TEST_F(TestBoostPredictor, TestEdgeCaseHighValue) {
    int result = predictor->predict_class(10, 20, 30, 40, 50, 0, 0, 1, 1, 1, 60);
    ASSERT_EQ(result, 1);
}

TEST_F(TestBoostPredictor, TestInvalidInputs) {
    ASSERT_THROW(
        predictor->predict_class(1, 2, 3, std::numeric_limits<float>::infinity(), 5, 0, 1, 1, 1, 1, 4),
        std::invalid_argument
    );
    ASSERT_THROW(
        predictor->predict_class(1, 2, 3, 4, 5, 0, 2, 1, 1, 1, 4),
        std::invalid_argument
    );
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: manual/data/java/tests/test_catboost_predict.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_catboost_predict {

    @Test
    public void test_basic_classification() {
        function_catboost_predict predictor = new function_catboost_predict();
        int result = predictor.predict_class(1f, 2f, 3f, 4f, 5f, 0, 0, 1, 1, 1, 3f);
        assertEquals(1, result);
    }

    @Test
    public void test_custom_threshold() {
        function_catboost_predict predictor = new function_catboost_predict();
        int result = predictor.predict_class(10f, 15f, 20f, 25f, 30f, 0, 0, 1, 1, 1, 18f);
        assertEquals(1, result);
    }

    @Test
    public void test_edge_case_low_value() {
        function_catboost_predict predictor = new function_catboost_predict();
        int result = predictor.predict_class(5f, 6f, 7f, 8f, 9f, 0, 0, 1, 1, 1, 4f);
        assertEquals(0, result);
    }

    @Test
    public void test_edge_case_high_value() {
        function_catboost_predict predictor = new function_catboost_predict();
        int result = predictor.predict_class(10f, 20f, 30f, 40f, 50f, 0, 0, 1, 1, 1, 60f);
        assertEquals(1, result);
    }

    @Test
    public void test_invalid_inputs() {
        function_catboost_predict predictor = new function_catboost_predict();

        // 测试无效的 float 值
        assertThrows(IllegalArgumentException.class, () -> {
            predictor.predict_class(1f, 2f, 3f, Float.NaN, 5f, 0, 1, 1, 1, 1, 4f);
        });

        // 测试无效的整数标签（非 0 或 1）
        assertThrows(IllegalArgumentException.class, () -> {
            predictor.predict_class(1f, 2f, 3f, 4f, 5f, 0, -1, 1, 1, 1, 4f);
        });
    }
}